docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-modal.md

```diff
@@ -0,0 +1,154 @@
+   ```sh
+   dlt init sql_database snowflake
+   ```
+2. Open the file and define the Modal Image you want to run `dlt` in:
+   ```py
+   import dlt
+   import pendulum
+
+   from sql_database import sql_database, ConnectionStringCredentials, sql_table
+
+   import modal
+   import os
+
+   image = (
+       modal.Image.debian_slim()
+       .apt_install(["libpq-dev"]) # system requirement for postgres driver
+       .pip_install(
+           "sqlalchemy>=1.4", # how `dlt` establishes connections
+           "dlt[snowflake]>=0.4.11",
+           "psycopg2-binary", # postgres driver
+           "dlt[parquet]",
+           "psutil==6.0.0", # for `dlt` logging
+           "connectorx", # creates arrow tables from database for fast data extraction
+       )
+   )
+
+   app = modal.App("dlt-postgres-pipeline", image=image)
+   ```
+
+3. Wrap the provided `load_table_from_database` with the Modal Function decorator, Modal Secrets containing your database credentials, and a daily cron schedule
+   ```py
+   # Function to load the table from the database, scheduled to run daily
+   @app.function(
+       secrets=[
+           modal.Secret.from_name("snowflake-secret"),
+           modal.Secret.from_name("postgres-read-replica-prod"),
+       ],
+       # run this pipeline daily at 6:24 AM
+       schedule=modal.Cron("24 6 * * *"),
+       timeout=3000,
+   )
+   def load_table_from_database(
+       table: str,
+       incremental_col: str,
+       dev: bool = False,
+   ) -> None:
+       # Placeholder for future logic
+       pass
+   ```
+
+4. Write your `dlt` pipeline:
+   ```py
+   # Modal Secrets are loaded as environment variables which are used here to create the SQLALchemy connection string
+   pg_url = f'postgresql://{os.environ["PGUSER"]}:{os.environ["PGPASSWORD"]}@localhost:{os.environ["PGPORT"]}/{os.environ["PGDATABASE"]}'
+   snowflake_url = f'snowflake://{os.environ["SNOWFLAKE_USER"]}:{os.environ["SNOWFLAKE_PASSWORD"]}@{os.environ["SNOWFLAKE_ACCOUNT"]}/{os.environ["SNOWFLAKE_DATABASE"]}'
+   # Create a pipeline
+   schema = "POSTGRES_DLT_DEV" if dev else "POSTGRES_DLT"
+   pipeline = dlt.pipeline(
+       pipeline_name="task",
+       destination=dlt.destinations.snowflake(snowflake_url),
+       dataset_name=schema,
+       progress="log",
+   )
+   credentials = ConnectionStringCredentials(pg_url)
+   # defines the postgres table to sync (in this case, the "task" table)
+   source_1 = sql_database(credentials, backend="connectorx").with_resources("task")
+   # defines which column to reference for incremental loading (i.e. only load newer rows)
+   source_1.task.apply_hints(
+       incremental=dlt.sources.incremental(
+           "enqueued_at",
+           initial_value=pendulum.datetime(2024, 7, 24, 0, 0, 0, tz="UTC"),
+       )
+   )
+
+    # if there are duplicates, merge the latest values
+   info = pipeline.run(source_1, write_disposition="merge")
+   print(info)
+   ```
+
```

Comment:
Looks like the next step is missing: how this code ends up on Modal? How to trigger runs?

Reply:
This runs the pipeline once, but might be worth adding that you need to run `modal deploy` to actually schedule the pipelin.